### PR TITLE
Small fixes from building up another demo.

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Fix `Excel_Workbook.sheet` and add a test.
- Add icon for `Table.row_count` and `DB_Table.row_count`.
- Make `join_kind` widget `Display.Always`.
- Add expression as an option to `Aggregate_Column`.
- Add `Simple_Calculation.Copy` to create a copy.
- Add defaults to `Simple_Expression` so less errory.
- Set period to default to day for `date_diff` allowing use in expressions.
- Add `Text_Left`, `Text_Right`, `Text_Length` and `Format` to `Simple_Expression`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Simple_Expression.enso [hunk 1]
```diff
@@ -43,6 +44,10 @@ type Simple_Expression
             Simple_Calculation.And rhs -> input_column && (table.resolve rhs)
             Simple_Calculation.Or rhs -> input_column || (table.resolve rhs)
             Simple_Calculation.Trim where what -> input_column.trim where (table.resolve what)
+            Simple_Calculation.Text_Left length -> input_column.text_left (table.resolve length)
+            Simple_Calculation.Text_Right length -> input_column.text_right (table.resolve length)
+            Simple_Calculation.Text_Length -> input_column.length
```
Comment (AdRiley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Simple_Expression.enso]:
```suggestion
            Simple_Calculation.Text_Length -> input_column.text_length
```

**Review by AdRiley — APPROVED** [review]
One small fix

File: distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso [hunk 2]
```diff
@@ -1546,7 +1546,7 @@ type DB_Column
          unusual events like DST.
     @period Date_Time_Helpers.make_period_selector_for_column
     date_diff : (DB_Column | Date | Date_Time | Time_Of_Day) -> Date_Period | Time_Period -> DB_Column ! Invalid_Value_Type | Illegal_Argument
-    date_diff self end (period : Date_Period | Time_Period) =
+    date_diff self end (period : Date_Period | Time_Period = Time_Period.Day) =
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: Note the peculiarity of `Time_Period.Day` (it is equal to 24h, not a calendar day - 99% of the time these are equivalent, but sometimes they are not):
image

I'd prefer to use `Date_Period.Day` here instead.

Comment (jdunkerley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso]: Changed to this.

**Review by radeusgd — APPROVED** [review]

**Pull request merged**